server/src/test/java/org/solarbank/server/integration/EnergySavingControllerIT.java

```diff
@@ -0,0 +1,76 @@
+package org.solarbank.server.integration;
+
+import org.junit.jupiter.api.Test;
+import org.solarbank.server.EnergySavingController;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@AutoConfigureMockMvc
+public class EnergySavingControllerIT extends IntegrationTestBase {
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Test
+    public void energySavingController_validInput_status200() throws Exception {
```

Comment:
this is my attempt at an integration test for valid user input. Not sure if I have got the right idea here